src/core/Akka.Remote/Serialization/DaemonMsgCreateSerializer.cs

```diff
@@ -114,6 +114,17 @@ public override object FromBinary(byte[] bytes, Type type)
         {
             var proto = DaemonMsgCreateData.ParseFrom(bytes);
             var clazz = Type.GetType(proto.Props.Clazz);
+
+            if (clazz == null)
+            {
+                var msg =
+                    string.Format(
+                        "Could not find '{0}' in loaded assemblies. If this is a remote/cluster deploy scenario, " +
+                        "ensure the remote system has referenced the assembly containing this type.", proto.Props.Clazz);
+                    
+                throw new ActorInitializationException(msg);
```

Comment:
> If we just check for null, we will not get the real inner type load exception.
I figure it would be good to actually get that specific exception, even if it is just an inner exception for the user, as that is the "real" cause of the problem.

I vote to do this:

```
Type.GetType(..., true)
```

And throw a `TypeLoadException`, rather than add a custom exception. Reduce clutter, get the specific details, and we can customize the error message to indicate that this was an issue with remote deployment. Agreed?